File: ecomm.services/src/main/java/com/joaoigm/ecomm/services/models/product/Product.java
package com.joaoigm.ecomm.services.models.product;

import com.fasterxml.jackson.annotation.JsonInclude;

import javax.persistence.*;
import java.util.Currency;

@Entity
@Table(name = "products")
public class Product {

    @javax.persistence.Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;
    @Column(length = 60)
    private String name;
    @Column(length = 500)
    private String description;
    private Double price;
    private Integer stock;

    @Transient
    @JsonInclude
    private Boolean Available;

    public Product(String name, String description, Double price, Integer stock) {
        this.name = name;
        this.description = description;
        this.price = price;
        this.stock = stock;
    }

    public Product(){}

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Double getPrice() {
        return price;
    }

    public void setPrice(Double price) {
        this.price = price;
    }

    public Integer getStock() {
        return stock;
    }

    public void setStock(Integer stock) {
        this.stock = stock;
    }

    public Boolean getAvailable() {
        return this.stock > 0;
    }

    public Long getId() {
        return this.id;
    }
}

File: ecomm-front/src/app/models/product.ts
export default class Product {

    
    private _id : number;
    public get id() : number {
        return this._id;
    }
    
    
    private _name : string;
    public get name() : string {
        return this._name;
    }
    public set name(v : string) {
        this._name = v;
    }


    
    private _description : string;
    public get description() : string {
        return this._description;
    }
    public set description(v : string) {
        this._description = v;
    }
    
    
    private _price : number;
    public get price() : number {
        return this._price;
    }    
    
    private _stock : number;
    public get stock() : number {
        return this._stock;
    }

    private _available : boolean;
    public get available() : boolean {
        return this._available;
    }

    private _selectedQuantity : number;
    public get selectedQuantity() : number {
        return this._selectedQuantity;
    }
    public set selectedQuantity(v : number) {
        this._selectedQuantity = v;
    }
    

    constructor(id: number, name: string, description:string, price:number, stock:number, available:boolean){
        this._id = id;
        this._name = name;
        this._description = description;
        this._price = price;
        this._stock = stock;
        this._available = available;
    }
}